v3/src/components/graph/adornments/movable-line.tsx

```diff
@@ -206,7 +224,25 @@
           newIntercept = isVertical ? pivot1.x : (newPivot2.y - newSlope * newPivot2.x)
         model.setLine({slope: newSlope, intercept: newIntercept, pivot1, pivot2: newPivot2}, lineKey)
       }
-    }, [lineKey, model, xScaleCopy, yScaleCopy])
+    }, [lineKey, model, xScaleCopy, yScaleCopy]),
+
+    moveEquation = useCallback((event: { x: number, y: number, dx: number, dy: number }) => {
+      if (event.dx !== 0 || event.dy !== 0) {
+        const equationContainerSelector = setEquationContainerSelector(),
+          equation = select(`${equationContainerSelector} p`).node() as Element,
+          equationWidth = equation?.getBoundingClientRect().width || 0,
+          equationHeight = equation?.getBoundingClientRect().height || 0,
+          left = event.x - equationWidth / 2,
+          top = event.y - equationHeight / 2
+
+        select(`${equationContainerSelector} p`)
```

Comment:
Whether the equation is pinned or not is part of the model state since when the document is saved and restored we don't want a pinned equation to start moving when the line is dragged.